ovn-tester/ovn_tester.py

```diff
@@ -178,41 +187,83 @@ def configure_tests(yaml, central_node, worker_nodes, global_cfg):
         mod = importlib.import_module(f'tests.{section}')
         class_name = ''.join(s.title() for s in section.split('_'))
         cls = getattr(mod, class_name)
-        tests.append(cls(yaml, central_node, worker_nodes, global_cfg))
+        tests.append(cls(yaml, central_nodes, worker_nodes, global_cfg))
     return tests
 
 
 def create_nodes(cluster_config, central, workers):
-    mgmt_net = cluster_config.node_net
-    mgmt_ip = mgmt_net.ip + 2
-    internal_net = cluster_config.internal_net
-    external_net = cluster_config.external_net
-    gw_net = cluster_config.gw_net
-    db_containers = (
-        ['ovn-central-1', 'ovn-central-2', 'ovn-central-3']
-        if cluster_config.clustered_db
-        else ['ovn-central']
-    )
+    node_az_conf = [
+        NodeConf(
+            cluster_config.node_net,
+            DualStackSubnet.next(
+                cluster_config.internal_net,
+                i * (cluster_config.n_workers // cluster_config.n_az),
+            ),
+            DualStackSubnet.next(
+                cluster_config.external_net,
+                i * (cluster_config.n_workers // cluster_config.n_az),
+            ),
+            DualStackSubnet.next(
+                cluster_config.gw_net,
+                i * (cluster_config.n_workers // cluster_config.n_az),
+            ),
+            cluster_config.ts_net,
+        )
+        for i in range(cluster_config.n_az)
+    ]
+
+    db_containers = [
+        [
+            f'ovn-central-az{i+1}-1',
+            f'ovn-central-az{i+1}-2',
+            f'ovn-central-az{i+1}-3'
+            if cluster_config.clustered_db
+            else f'ovn-central-az{i+1}',
+        ]
+        for i in range(cluster_config.n_az)
+    ]
+
     relay_containers = [
-        f'ovn-relay-{i + 1}' for i in range(cluster_config.n_relays)
+        [f'ovn-relay-az{i+1}-{j+1}' for j in range(cluster_config.n_relays)]
+        for i in range(cluster_config.n_az)
     ]
-    central_node = CentralNode(
-        central, db_containers, relay_containers, mgmt_net, mgmt_ip
-    )
-    worker_nodes = [
-        WorkerNode(
+
+    central_nodes = [
+        CentralNode(
+            central,
```

Comment:
This will always start the central OVN nodes (ovn-northd + NB + SB) on the physical node that's designated as "central" in the physical deployment file.

That's not exactly enough to simulate real deployments.  What we need is the ability to specify a list of physical nodes that may run central components.  For example:
```
central-nodes:
  - foo1.somewhere.else
  - foo2.somewhere.else

worker-nodes:
  - bar1.somewhere.else
  - bar2.somewhere.else
  - bar3.somewhere.else
 ```

Now, if we run ovn-heater with a test file that needs to run n_az == 2 then each of the two central nodes, should be running central components:
- foo1: northd + NB + SB AZ1
- foo2: northd + NB + SB AZ2

If we need to simulate more AZs than central nodes, then we should distribute those equally across central-nodes.  E.g., n_az == 3:
- foo1: northd + NB + SB AZ1
- foo2: northd + NB + SB AZ2
- foo1: northd + NB + SB AZ3

Note: I think it's probably OK to break backwards compatibility in this case.

Reply:
@dceara ack, do you agree to add this feature on top of the current series?

Reply:
Ack.